docs/slicing.md

```diff
@@ -0,0 +1,145 @@
+# Slicing
+
+Functions (which includes strings, arrays, etc.) can be called using slice syntax, The syntax is as follows:
+
+```text
+someSet(optionalLowerBoundExpr;optionalUpperBoundExpr;optionalStepSizeExpr)
+```
+
+As shown above, you can define a slice by defining a lower bound, an upper bound, both or none.
+You can also define a `stepSize` that is used to define the increment of the numbers from the lower
+bound to the upper bound. `stepSize` defaults to `1` when it is not defined.
+
+## Usages
+
+Only certain types of sets can be sliced. These sets are:
+
+1. `Array`
+2. `String`
+3. `Byte Array`
+4. `Dictionary`
+
+Example usage:
+
+| expression | equals |
+|:-|:-|
+| `[1, 2, 3, 4, 5](1;4)` | `[2, 3, 4]` |
+| `"this is a sentence"(;7)` | `"this is"` |
+| `{1: "hello", 2: "there", 3: 123, 4: 456}(;4)` | `["hello", "there", 123]` |
+| `"sentence"(;)` | `"sentence"` |
+| `[1, 2, 3, 4, 5](1;5;2)` | `[2, 4]` |
+| `[1, 2, 3, 4, 5](3;0;-1)` | `[4, 3, 2]` |
+| `[1, 2, 3, 4, 5](;1;-1)` | `[5, 4, 3]` |
+| `[1, 2, 3, 4, 5](3;;-1)` | `[4, 3, 2, 1]` |
+| `[1, 2, 3, 4, 5](;;-1)` | `[5, 4, 3, 2, 1]` |
+
+Generic sets (i.e. unindexed sets) or any other expression that does not evaluate to the
+special types of sets will fail the expression.
+
+An example of an unindexed sets is as follows:
+
+```text
+{1, 2, 3, "abc"}
+```
+
+Since slice is also an argument, the following expression is also valid
+
+```text
+[1, 2, 3, 4, 5](;;-1, 0)
+```
+
+this expression reverses the array and fetches the first value. In
+this example, it will evaluate to `5`.
+
+### Behaviour
+
+#### Default values
+
+Slice operation behaves differently when provided with different expressions.
+
+Lower bound is always inclusive while upper bound is always exclusive. As in it
+always includes values that corresponds to ranges of values from `start` to
+`end - 1`. This is true for all scenarios, whether `start` or `end` are defined or not.
+However, when `end` is not provided, the last element is always included. This is not
+a special case as default value of `end` will include the last element. Refer to the
+explanations below.
```

Comment:
If this mirrors current behaviour, I'm happy to document it as is. However, `end` should be a special case, since step might take on odd values, like `0.01`, which would break the described semantic. If necessary, raise an issue to rectify this and link it here.

Reply:
current implementation doesn't take into account for odd values of `end`. Will raise an issue